File: scripts/python/parsers/layout_parser.py
"""
layout_parser.py
Reads Salesforce Page Layout metadata from SFDX source format.
Uses stdlib only (pathlib).
"""

from pathlib import Path


def parse_layouts(layout_dir: str) -> dict:
    """
    Parse all layouts under layout_dir.
    Returns {'layouts': [{'name'}]}
    """
    layouts = []
    p = Path(layout_dir)

    if not p.exists():
        return {'layouts': layouts}

    for ff in sorted(p.glob('*.layout-meta.xml')):
        name = ff.name.replace('.layout-meta.xml', '')
        layouts.append({'name': name})

    return {'layouts': layouts}

File: scripts/node/parsers/layoutParser.js
/**
 * layoutParser.js
 * Reads Salesforce Page Layout metadata from SFDX source format.
 */

'use strict';

const fs   = require('fs');
const path = require('path');

/**
 * @param {string} layoutDir  - path to the layouts/ folder
 * @returns {{ layouts: Array<{ name }> }}
 */
async function parseLayouts(layoutDir) {
  const layouts = [];

  if (!fs.existsSync(layoutDir)) return { layouts };

  const files = fs.readdirSync(layoutDir).filter(f => f.endsWith('.layout-meta.xml'));

  for (const file of files) {
    const name = file.replace('.layout-meta.xml', '');
    layouts.push({ name });
  }

  return { layouts };
}

module.exports = { parseLayouts };
